curriculum/roadmap/llm.py

```
@@ -61,7 +72,25 @@ def generate(self, prompt, deadline=None):
+                return self._call(model, body, deadline)
+            except LLMError as e:
+                last = e
+                # 429 만 갈아탄다. 한도는 모델별로 따로 차기 때문이다.
+                # 5xx 나 네트워크 오류는 모델을 바꿔도 그대로라 여기서 끝낸다.
+                if e.status != 429:
+                    raise
+                if deadline is not None and time.monotonic() >= deadline:
+                    raise
+        raise last
+
+    def _call(self, model, body, deadline):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_call`

(ANN202)